src/Collection/CollectionInterface.php

```diff
@@ -783,6 +783,16 @@ public function jsonSerialize();
      */
     public function compile($preserveKeys = true);
 
+    /**
+     * Returns a new collection where any operations chained after it are guaranteed
+     * to be run lazily. That is, elements will be yieleded one at a time.
+     *
+     * A lazy collection can only be iterated once. A second attempt results in an error.
```

Comment:
:+1:  Good note! This should be included in the Cookbook, too.

Reply:
Would there be any value in a test verifiing this behavior? (if we aren't doing that already)